internal/consensus/reactor.go

```diff
@@ -1101,8 +1104,10 @@ func (r *Reactor) handleStateMessage(ctx context.Context, envelope *p2p.Envelope
 		ps.ApplyNewValidBlockMessage(msgI.(*NewValidBlockMessage))
 
 	case *tmcons.HasVote:
-		ps.ApplyHasVoteMessage(msgI.(*HasVoteMessage))
-
+		if err := ps.ApplyHasVoteMessage(msgI.(*HasVoteMessage)); err != nil {
+			r.logger.Error("peer sent us an invalid HasVote msg", "msg", msg, "err", err)
```

Comment:
```suggestion
			r.logger.Error("applying HasVote message", "msg", msg, "err", err)
```
The `err` value contains the specific info here.